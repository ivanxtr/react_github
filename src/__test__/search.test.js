import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../components/searchComponent.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('Search Component', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = create(<Search />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should render the proper structure', () => {
    const wrapper = mount(<Search />)
    expect(wrapper.find('input')).toHaveLength(1);
  })
})
