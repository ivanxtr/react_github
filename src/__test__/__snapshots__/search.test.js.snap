// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component snapshot render 1`] = `
<input
  className="search"
  onChange={[Function]}
  placeholder="Search"
  type="search"
/>
`;
